litestar/_openapi/responses.py

```diff
@@ -69,24 +75,29 @@ def create_success_response(  # noqa: C901
     schemas: dict[str, Schema],
 ) -> OpenAPIResponse:
     """Create the schema for a success response."""
-    return_annotation = route_handler.parsed_fn_signature.return_type.annotation
+    return_type = route_handler.parsed_fn_signature.return_type
+    return_annotation = return_type.annotation
     default_descriptions: dict[Any, str] = {
-        Stream: "Stream Response",
-        Redirect: "Redirect Response",
-        File: "File Download",
+        StreamingResponse: "Stream Response",
+        RedirectResponse: "Redirect Response",
+        FileResponse: "File Download",
     }
     description = (
         route_handler.response_description
         or default_descriptions.get(return_annotation)
         or HTTPStatus(route_handler.status_code).description
     )
 
-    if return_annotation not in {Signature.empty, None, NoneType, Redirect, File, Stream}:
-        if return_annotation is Template:
+    if return_annotation is not Signature.empty and not return_type.is_subclass_of(
+        (NoneType, FileResponse, RedirectResponse, StreamingResponse, ASGIResponse)
+    ):
+        if return_annotation is TemplateResponse:
             return_annotation = str
             route_handler.media_type = get_enum_string_value(MediaType.HTML)
-        elif is_class_and_subclass(get_origin(return_annotation), LitestarResponse):
-            return_annotation = get_args(return_annotation)[0] or Any
+        elif return_type.is_subclass_of(LitestarResponse):
+            return_annotation = return_type.inner_types[0].annotation if return_type.inner_types else Any
+            if not route_handler.media_type:
+                route_handler.media_type = get_enum_string_value(MediaType.JSON)
```

Comment:
We don't actually know what the media type should be here, as it can be dynamically set on the response class returned by the handler. Not much we can do though, except suggest that the user set the appropriate media type on the handler if they want the docs correct.